substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1230,7 +1230,11 @@ pub mod pallet {
 				.map(|t| T::Lookup::lookup(t).map_err(DispatchError::from))
 				.map(|n| {
 					n.and_then(|n| {
-						if old.contains(&n) || !Validators::<T>::get(&n).blocked {
+						// a good target nomination must be a valiator (active or idle). The
+						// validator must not be blocked.
+						if Self::status(&n).is_ok() &&
```

Comment:
Should we check here that the status is actually `Validator` or `Idle`, so no `Err` or `Nominator` allowed?